darwin/Platform.c

```diff
@@ -357,9 +381,80 @@ void Platform_getFileDescriptors(double* used, double* max) {
 }
 
 bool Platform_getDiskIO(DiskIOData* data) {
-   // TODO
-   (void)data;
-   return false;
+   if (!iokit_available)
+      return false;
+
+   io_iterator_t drive_list;
+
+   /* Get the list of all drives */
+   if (IOServiceGetMatchingServices(iokit_port, IOServiceMatching("IOBlockStorageDriver"), &drive_list))
+      return false;
+
+   unsigned long long int read_sum = 0, write_sum = 0, timeSpend_sum = 0;
+   io_registry_entry_t drive;
+   while ((drive = IOIteratorNext(drive_list)) != 0) {
+      CFMutableDictionaryRef properties_tmp = NULL;
+
+      /* Get the properties of this drive */
+      if (IORegistryEntryCreateCFProperties(drive, &properties_tmp, kCFAllocatorDefault, 0)) {
+         IOObjectRelease(drive);
+         IOObjectRelease(drive_list);
+         return false;
+      }
+
+      CFDictionaryRef properties = properties_tmp;
+
+      if (properties != NULL) {
```

Comment:
Why not invert this condition and flatten the block below.

```c
if (!properties) {
   continue;
}
```

You could even postpone the `CFMutableDictionaryRef` -> `CFDictionaryRef` conversion to after this check.